### ISSUE TITLE
check_syntax: false positive

### ISSUE THREAD

**Issue by g-back** [abaplint/abaplint#747]
```abap
gp_amount_sp = REDUCE #( 
                  INIT x TYPE vbrk-netwr
                  FOR <i> IN invoices
                  WHERE ( vtweg = '20' )
                  NEXT x = x + <i>-kwert ).
```
`"x" not found (check_syntax)`

**Comment by larshp** [comment]
`DATA(foo) = REDUCE i( INIT s TYPE i FOR i = 1 UNTIL i > 10 NEXT s = s + i ).`
